pkg/volume/csi/csi_client.go

```diff
@@ -69,14 +69,44 @@ type csiClient interface {
 
 // csiClient encapsulates all csi-plugin methods
 type csiDriverClient struct {
-	driverName string
-	nodeClient csipb.NodeClient
+	driverName        string
+	nodeClientCreator nodeClientCreator
 }
 
 var _ csiClient = &csiDriverClient{}
 
+type closer func() error
```

Comment:
Apologies for confusion.  There is a Go built-in `io.Closer` interface already.  Can you please retrofit your code to use that interface.  It has the exact signature that you are already using.  That way we're minimizing type sprawl.

See io.Closer interface - https://golang.org/pkg/io/#Closer